src/Symfony/Component/Security/Core/Exception/AuthenticationException.php

```diff
@@ -21,8 +21,17 @@
  */
 class AuthenticationException extends RuntimeException
 {
+    /** @internal */
+    protected $serialized;
+
     private $token;
 
+    public function __construct(string $message = '', int $code = 0, \Throwable $previous = null)
+    {
+        unset($this->serialized);
```

Comment:
I have the case when using phpspec in some unit test and trying to match if a `BadCredentialsException` is throwed (inherited from this one), that phpscpec test case crashes because phpspec do some reflection on this class and cannot find the property as it is unset.

I'm not sure if things like that can also happen with phpunit, but isn't the scope of the property enough?

Reply:
Ok forgot this comment I've just read https://github.com/symfony/symfony/pull/44037#issuecomment-975760499